node-graph/gstd/src/gpu_nodes.rs

```diff
@@ -64,7 +64,7 @@ impl<T: GpuExecutor> Clone for ComputePass<T> {
 }
 
 #[node_macro::node_impl(MapGpuNode)]
-async fn map_gpu<'a: 'input>(image: ImageFrame<Color>, node: DocumentNode, editor_api: graphene_core::application_io::EditorApi<'a, WasmApplicationIo>) -> ImageFrame<Color> {
+async fn map_gpu<'a: 'input>(image: ImageFrame<Color>, node: DocumentNode, editor_api: graphene_core::application_io::EditorApi<'a, WasmApplicationIo>) -> Result<ImageFrame<Color>, String> {
```

Comment:
Same argument as for the BlendGpu node should return an empty image frame as well

Reply:
Got it, thanks - should I do something similar with `compile_gpu` and `create_compute_pass_descriptor` or are those ok as they (as far as I understand) are mostly called within the user-facing functions like the ones I have amended?

Reply:
Yes, basically all nodes

Reply:
Cool - in that case I think my latest commit should be all of the nodes accounted for. 

`create_compute_pass_descriptor` is not a node as far as I can see and `compile_gpu` seems to be a compilation process that probably should be able to fail, much like the `compile` method in `graphene-compiler.rs`. It looks like, from the TODO on it, that `compile_gpu` is intended to be moved to `graph-craft`, I can have a look at doing that maybe on a different PR if this is still intended.